### PR TITLE
Append newline to the end of dump methods

### PR DESCRIPTION
**Pull request by omus**
The doctest system doesn't deal very well with examples that don't have a trailing newline. Originally mentioned in #17106 and in the [julia-dev forums](https://groups.google.com/forum/#!topic/julia-dev/uHpfvBkClTc)

This is my first time making modifications to the `dump` methods. Let me know if there's something I didn't consider.

### CODE REVIEW

File: base/show.jl [hunk 1]
```diff
@@ -1158,6 +1168,7 @@ function dumptype(io::IO, x::ANY, n::Int, indent)
             end
         end
     end
+    isempty(indent) && println(io)
 end
```
Comment (omus) [inline base/show.jl]: Maybe all the `dump` methods should explicitly return nothing?

**Comment by kshyatt** [comment]
LGTM. Anyone else have thoughts?

File: base/show.jl (hunk 1 above)
Comment (JeffBezanson) [inline base/show.jl]: Yes; otherwise they return `false` which doesn't really make sense.

**Comment by omus** [comment]
Would be nice to get this merged.

File: test/show.jl [hunk 2]
```diff
@@ -500,3 +500,9 @@ let s  = IOBuffer(Array{UInt8}(0), true, true)
     Base.showarray(s, [1,2,3], false, header = false)
     @test String(resize!(s.data, s.size)) == " 1\n 2\n 3"
 end
+
+# The `dump` function should alway have a trailing newline
```
Comment (tkelman) [inline test/show.jl]: always

Comment (omus) [inline test/show.jl]: Thanks. I'll take care of that in another PR

**Pull request merged**